Tutorials/creating-2d-metroidvania-game-with-splashkit/2-1-drawing-options-functions-using-bitmap.md

```diff
@@ -0,0 +1,171 @@
+---
+title: "Drawing Options functions using DrawBitmap"
+description: Learn how to use drawing options functions in Splashkit and combine them to create dynamic and visually stunning graphics.
+author: Sharvani Kandala
+date: 05/05/2024
+tags: programming, graphics, Splashkit, tutorials
+sidebar: 
+   label: " -2-1 Drawing Options Functions using Bitmap"
+---
+
+## Window Setup
+
+To begin utilizing drawing options functions in Splashkit, set up the game window as follows:
+
+1. Import the necessary Splashkit headers at the beginning of your code:
+
+   ```cpp
+   #include "splashkit.h"
+   ```
+1. In the `main` function, configure the game window and any necessary resources:
+
+   ```cpp
+    int main()
+    {
+       open_window("Advanced Drawing", 800, 600);
+       load_resource_bundle("game_resources", "r");
+       return 0;
+    }
+    ```
+## Drawing Options Functions
+
+## Example 1: Applying Option Part Bmp:
+
+In this example, we'll demonstrate how to use 'Option Part Bmp' to draw a section of a bitmap, effectively cropping the image.
+
+# Function Syntax:
+
+ ```cpp
+ void draw_bitmap(bitmap bmp, double x, double y, drawing_options opts);
+ ```
+
+ - `bmp`: The bitmap to draw.
+ - `x`: The x-coordinate for the top-left corner of the bitmap.
+ - `y`: The y-coordinate for the top-left corner of the bitmap.
+ - `opts`: Drawing options to apply.
+
+1. **Example Usage:**
+
+   For instance, draw a window with dimensions of `800` pixels width and `600` pixels height. Here is the code snippet where we load a bitmap and applies 'Option Part Bmp' to draw only a 
+   100x100 region of it at coordinates (100, 100). The function `load_resource_bundle` loads a resource bundle named `game_resources` from the specified folder `resource_folder`.
+   Resource bundles typically contain images, sounds, and other assets used in the game.`bitmap bmp` loads a bitmap image named `my_bitmap` from the file `Resources/images/bitmap.png`.
+   The bitmap is stored in the variable bmp for later use.
+
+   ```cpp
+   int main()
+    {
+       open_window("Drawing", 800, 600);
+       load_resource_bundle("game_resources", "resource_folder");
+
+       bitmap bmp = load_bitmap("my_bitmap", "Resources/images/bitmap.png");
```

Comment:
The second argument passed to load_bitmap should just be the filename, no need for `Resources/images/`.